### PR TITLE
Fix issues with row limit trigger and truncate table

### PR DESCRIPTION
**Pull request by wolffcm**
When writing the example app I found a few issues that had to do with truncating a table:
* When truncating a table, the row limit trigger for a table would disappear.  It needs to be propagated to the new PersistentTable object that gets created.
* When the row limit trigger itself does a truncate table, then the insert executor's pointer to the PersistentTable being modified needs to be updated; otherwise we're just attempting to insert into the old, full table.

### CODE REVIEW

File: src/ee/executors/insertexecutor.cpp [hunk 1]
```diff
@@ -270,7 +282,7 @@ bool InsertExecutor::p_execute(const NValueArray &params) {
                 // try to put the tuple into the target table
 
                 if (m_hasPurgeFragment) {
-                    if (!executePurgeFragmentIfNeeded(persistentTable))
+                    if (!executePurgeFragmentIfNeeded(&persistentTable))
```
Comment (jxchinaren) [inline src/ee/executors/insertexecutor.cpp]: oh... Does this mean that UPSERT may delete a tuple when UPSERT is really a UPDATE? If so, this does not seem correctly.

Comment (wolffcm) [inline src/ee/executors/insertexecutor.cpp]: No, this block is entered when we're inserting a row.  The test to see if we found a PK match is line 281.  You had me scared for a second!

File: src/ee/executors/insertexecutor.cpp [hunk 2]
```diff
@@ -151,6 +152,13 @@ bool InsertExecutor::executePurgeFragmentIfNeeded(PersistentTable* table) {
                        tupleLimit);
             return false;
         }
+
+        // If the purge fragment did a truncate table, then the old
+        // table is still around for undo purposes, but there is now a
+        // new empty table we can insert into.  Update the caller's table
+        // pointer to use it.
+        TableCatalogDelegate* tcd = m_engine->getTableDelegate(table->name());
```
Comment (jxchinaren) [inline src/ee/executors/insertexecutor.cpp]: Instead of caching the variable PersistentTable* table and use its name to look up the new table, you can just call function "m_node->getTargetTable()" to get the new table. 

The node has the delegate already and can bypass the treemap look up.

File: src/ee/executors/insertexecutor.cpp (hunk 1 above)
Comment (jxchinaren) [inline src/ee/executors/insertexecutor.cpp]: I see now. That will make it correctly. 哈哈, thanks.

**Comment by jxchinaren** [comment]
Finish this review round.

File: src/ee/executors/insertexecutor.cpp (hunk 1 above)
Comment (jxchinaren) [inline src/ee/executors/insertexecutor.cpp]: One more question: do we have UPSERT tests with capped table feature?

Comment (wolffcm) [inline src/ee/executors/insertexecutor.cpp]: Yes, there is a simple test for it in TestSQLFeaturesNewSuite.  It just makes sure that we will delete for inserts, and that we don't delete any rows (even if table is full) for updates.  Maybe there should be more?

**Comment by wolffcm** [comment]
I just pushed a new commit that addresses the map lookup issue.

File: src/ee/executors/insertexecutor.cpp (hunk 1 above)
Comment (jxchinaren) [inline src/ee/executors/insertexecutor.cpp]: It's better to have that test, but I am also ok to merge now. It's an edge case and separate from this pull request ticket.

**Comment by jxchinaren** [comment]
Looks good to me.

File: src/ee/executors/insertexecutor.cpp (hunk 1 above)
Comment (wolffcm) [inline src/ee/executors/insertexecutor.cpp]: Ok, thanks Xin.  I can file a ticket for testing UPSERT with ROW LIMIT triggers.
